File: NovaCore-Spigot_v1_19_R1/src/main/java/net/zeeraa/novacore/spigot/version/v1_19_R1/VersionIndependentLoader.java
package net.zeeraa.novacore.spigot.version.v1_19_R1;

import net.zeeraa.novacore.spigot.abstraction.CommandRegistrator;
import net.zeeraa.novacore.spigot.abstraction.Listeners;
import net.zeeraa.novacore.spigot.abstraction.VersionIndependentUtils;

public class VersionIndependentLoader implements net.zeeraa.novacore.spigot.abstraction.VersionIndependantLoader {
	@Override
	public CommandRegistrator getCommandRegistrator() {
		return new net.zeeraa.novacore.spigot.version.v1_19_R1.CommandRegistrator();
	}

	@Override
	public VersionIndependentUtils getVersionIndependentUtils() {
		return new net.zeeraa.novacore.spigot.version.v1_19_R1.VersionIndependentUtils();
	}

	@Override
	public Listeners getListeners() {
		return new net.zeeraa.novacore.spigot.version.v1_19_R1.Listeners();
	}
}

File: NovaCore-Spigot_v1_19_R1/src/main/kotlin/net/zeeraa/novacore/spigot/version/v1_19_R1/VersionIndependentLoader.kt
package net.zeeraa.novacore.spigot.version.v1_19_R1

import net.novauniverse.spigot.version.shared.v1_16plus.NativeParticleImplementation
import net.zeeraa.novacore.spigot.abstraction.CommandRegistrator
import net.zeeraa.novacore.spigot.abstraction.Listeners
import net.zeeraa.novacore.spigot.abstraction.VersionIndependantLoader
import net.zeeraa.novacore.spigot.abstraction.VersionIndependentUtils
import net.zeeraa.novacore.spigot.abstraction.particle.NovaParticleProvider

class VersionIndependentLoader : VersionIndependantLoader() {
	override fun getCommandRegistrator(): CommandRegistrator {
		return NMSBasedCommandRegistrator()
	}

	override fun getVersionIndependentUtils(): VersionIndependentUtils {
		return VersionIndependentUtilsImplementation(this)
	}

	override fun getListeners(): Listeners {
		return ListenersImplementation()
	}

	override fun getVersionSpecificParticleProvider(): NovaParticleProvider {
		return NativeParticleImplementation()
	}
}